streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -178,6 +184,67 @@ public void register(final StateStore store,
         stores.put(store.name(), store);
     }
 
+    public void reinitializeStateStoresForPartitions(final Collection<TopicPartition> partitions,
```

Comment:
One more thing... `partitions` is always just a singleton collection. Can we save some complexity by changing the parameter to `TopicPartition`?

Reply:
Ack -- I guess this is due to back porting -- in `trunk` it can be multiple. Will fix in this PR.